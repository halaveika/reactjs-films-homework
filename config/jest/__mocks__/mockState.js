const mockState = {
  content: {
    results: [{
      id: 2412412,
      title: 'Fight club',
      genre_ids: [13, 20, 21],
      vote_average: 8,
      poster_path: 'image_path',
      overview: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry',
    },
    {
      id: 41412,
      title: 'Sopranos',
      genre_ids: [13, 18],
      vote_average: 10,
      poster_path: 'image_path',
      overview: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry',
    }],
    page: 1,
    total_pages: 12,
    total_results: 55,
    genres_array: [{
      id: 13,
      name: 'Adventure',
    }, {
      id: 18,
      name: 'Drama',
    }, {
      id: 20,
      name: 'Family',
    }, {
      id: 21,
      name: 'Fantasy',
    }],
    video_url: 'test_url',
    details: {
      id: 2412412,
      title: 'Fight club',
      runtime: 120,
      vote_average: 10,
      poster: 'image_path',
      overview: 'Lorem Ipsum is simply dummy text of the printing',
    },
    isInitialisated: false,
  },
};

export default mockState
