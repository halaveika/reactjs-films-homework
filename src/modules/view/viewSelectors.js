
export const getListFlexDirectionView = (state) => state.view.isRow;

export const getGenreView = (state) => state.view.genre;


