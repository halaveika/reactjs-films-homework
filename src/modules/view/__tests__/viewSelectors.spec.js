import { getListFlexDirectionView, getGenreView } from '../viewSelectors';
import mockState from '../../../../config/jest/__mocks__/mockState';

describe('contentSelectors testing', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  test('getListFlexDirectionView should return isRow from state', () => {
    expect(getListFlexDirectionView(mockState)).toMatchSnapshot();
  });

  test('getGenreView should return genre from state', () => {
    expect(getGenreView(mockState)).toMatchSnapshot();
  });
});
