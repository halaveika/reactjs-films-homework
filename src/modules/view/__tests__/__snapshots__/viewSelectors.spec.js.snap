// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`contentSelectors testing getGenreView should return genre from state 1`] = `""`;

exports[`contentSelectors testing getListFlexDirectionView should return isRow from state 1`] = `true`;
